elrond-wasm/src/types/interaction/arg_buffer.rs

```diff
@@ -34,6 +34,16 @@ impl ArgBuffer {
 		self.arg_data.as_ptr()
 	}
 
+	/// returns a clone of the raw arg data
+	pub fn arg_data(&self) -> Vec<u8> {
+		self.arg_data.clone()
+	}
+
+	/// returns a clone of the raw arg data lengths
+	pub fn arg_lengths(&self) -> Vec<usize> {
```

Comment:
Better return as slice &[usize] and let the whoever uses the method to .clone() is necessary